src/helper/args/mod.rs

```diff
@@ -98,19 +96,30 @@ pub fn get_args_help<Output: Write>(
         if args.is_empty() {
             return Ok(());
         }
-        for arg_variants in [
-            (config.check_version).then(|| &args[0]),
-            (config.check_help && args.len() >= 2).then(|| &args[1]),
-        ]
+        for arg_variants in if args[0].is_empty() {
+            [(args.len() > 1).then(|| &args[1..]), None]
```

Comment:
I feel like this can be simplified. It is very hard to understand what this is doing at first glance 💫